src/views/AddItem.jsx

```diff
@@ -3,30 +3,50 @@ import { Button, FormInput } from '../components';
 import { Link } from 'react-router-dom';
 import { addItem } from '../api';
 
-export function AddItem({ listToken }) {
+export function AddItem({ listToken, itemList }) {
 	const [formFields, setFormFields] = useState({
 		itemName: '',
 		daysUntilNextPurchase: 7,
 	});
 
-	const [message, setMessage] = useState('');
+	
+	const [message, setMessage] = useState(''); 
 
 	const { itemName, daysUntilNextPurchase } = formFields;
 
 	const handleSubmit = async (e) => {
 		e.preventDefault();
 
+		const itemNamesNoChar = itemList.map((item) =>
+			item.name.toLowerCase().replace(/[^a-zA-Z0-9 ]/g, ''),
+		); // Item Names without special characters
+
```

Comment:
1. I don't understand what the variable `itemNamesNoChar` is supposed to do. What is `no char`? I think you can also refer to them as `items` rather than `itemNames`
2. Why do you need regex in this case?
3. regarding the comment `// Item Names without special characters`, what is the reason you're removing special characters? If you allow them add special characters when adding the item, why would you remove them here? I believe the criteria is to compare irrespective of the letter casing, rather than special characters. If you don't want special characters, would you consider removing them before saving them to the database? 
4. I'd suggest you put the logic to validate the input in separate functions. The code here is already getting tangled up with a lot of nested conditions.

Reply:
ItemNames is iterating though the items array and we used a regex to remove special characters to make a comparison. We thought it is an easy way to implement a solution for text-processing cases